app/src/main/java/app/musikus/services/SessionService.kt

```diff
@@ -107,129 +109,112 @@ class SessionService : Service() {
-            initialDelay = timerInterval.inWholeMilliseconds,
-            period = timerInterval.inWholeMilliseconds
+            initialDelay = 0,
+            period = 1000.milliseconds.inWholeMilliseconds
         ) {
-            if (sessionState.value.isPaused) {
-                sessionState.update { it.copy(
-                    pauseDuration = it.pauseDuration + timerInterval)
-                }
+            updateNotification()
+        }
+    }
+
+    private fun togglePause() {
+        applicationScope.launch {
+            if (useCases.getPausedState()) {
+                useCases.resume()
             } else {
-                sessionState.update { it.copy(
-                    currentSectionDuration = it.currentSectionDuration + timerInterval)
-                }
+                useCases.pause()
             }
-            updateNotification()
+        }
+        updateNotification()
+    }
+
+
+    private fun newSection(item: LibraryItem) {
+        startTimer()
+        applicationScope.launch {
+            try {
+                useCases.selectItem(item)
+            } catch (e: IllegalStateException) {
+                Log.e("TAG", "Cannot start new section: ${e.message}")
+            }
+        }
+        updateNotification()
+    }
+
+    private fun removeSection(sectionId: UUID) {
+        applicationScope.launch {
+            useCases.deleteSection(sectionId)
         }
     }
 
+
+    /** -------------------------------------------- Service Boilerplate -------------------------*/
+
     private fun updateNotification() {
-        val mNotificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
-        mNotificationManager.notify(SESSION_NOTIFICATION_ID, createNotification())
+        applicationScope.launch {
+            val mNotificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
+            mNotificationManager.notify(SESSION_NOTIFICATION_ID, createNotification())
+        }
     }
 
     /**
      * Creates a notification object based on current session state
      */
-    private fun createNotification() : Notification {
-        val totalPracticeDuration = sessionState.value.sections.sumOf {
-            (it.duration ?: sessionState.value.currentSectionDuration).inWholeMilliseconds }.milliseconds
-        val totalPracticeDurationStr =
-            getDurationString(totalPracticeDuration, DurationFormat.HMS_DIGITAL)
-
-        val currentSectionName = sessionState.value.sections.lastOrNull()?.libraryItem?.name ?: "No Section"
-
-        val title: String
-        val description: String
-
-        if (sessionState.value.isPaused) {
-            title = "Practicing Paused"
-            description = "$currentSectionName - Total: $totalPracticeDurationStr"
-        } else {
-            title = "Practicing for $totalPracticeDurationStr"
-            description = currentSectionName
+    private suspend fun createNotification() : Notification {
```

Comment:
i'm pretty sure this function belongs in a use case... we could just write an issue tho

Reply:
Well, it's just getting some text to display in the notification, there is no actual calculation going on... But yes, we could write a usecase for it...